Task2 completed.

Sir-Beard opened the pull request:
Commit 14.12.2022 13:03

Sir-Beard committed:
Task completed.
Commit 14.12.2022 13:03

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,48 @@
+        createReport(toFileName);
+    }
+
+    private void readFile(String fromFileName) {
```
Этот метод выполняет 2 функции - чтение файла и подсчет данных. Это противоречит SRP.

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,48 @@
+    private void readFile(String fromFileName) {
+        try {
+            File csvFileReader = new File(fromFileName);
+            Scanner csvScanner = new Scanner(csvFileReader);
```
ресурсы нужно закрывать
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#try-with-resources)
try-with-resources

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,48 @@
+        }
+    }
 
+    private void createReport(String toFileName) {
```
Этот метод тоже выполняет две функции - записи в файл и формирования отчета.

Sir-Beard committed:
Task 2. WorkWithFile.
Edited code.
Commit 14.12.2022 23:20

Sir-Beard committed:
Task 2. WorkWithFile.
Edited code.
Commit 14.12.2022 23:30

Sir-Beard committed:
Task 2. WorkWithFile.
Edited code: with NIO and 3 methods in getStatistic method.
Commit 25.12.2022 18:39

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+        try {
+            return Files.readAllLines(path);
+        } catch (IOException e) {
+            throw new RuntimeException("File not found, " + e);
```
```suggestion
            throw new RuntimeException("Can not read file: " + fromFileName, e);
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+    }
+
+    private String createReport(List<String> dataFromFile) {
+        String lineOfDataFromFile;
+        String[] separatedData;
+        String operationType;
+        int amount;
```
```suggestion
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+        int sumSupply = 0;
+        int sumBuy = 0;
+        for (String line : dataFromFile) {
+            lineOfDataFromFile = line;
```
```suggestion
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+        int sumBuy = 0;
+        for (String line : dataFromFile) {
+            lineOfDataFromFile = line;
+            separatedData = lineOfDataFromFile.split(",");
+            operationType = separatedData[INDEX_OF_OPERATION];
+            amount = Integer.parseInt(separatedData[INDEX_OF_AMOUNT]);
```
```suggestion
            String[] separatedData = line.split(",");
            String operationType = separatedData[INDEX_OF_OPERATION];
            int amount = Integer.parseInt(separatedData[INDEX_OF_AMOUNT]);
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+            operationType = separatedData[INDEX_OF_OPERATION];
+            amount = Integer.parseInt(separatedData[INDEX_OF_AMOUNT]);
+            if (operationType.equals("supply")) {
+                sumSupply = sumSupply + amount;
```
```suggestion
                sumSupply += amount;
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+            if (operationType.equals("supply")) {
+                sumSupply = sumSupply + amount;
+            } else {
+                sumBuy = sumBuy + amount;
```
```suggestion
                sumBuy += amount;
```

Rommelua commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,59 @@
+        try {
+            Files.writeString(path, report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create file, " + e);
```
msg

Sir-Beard committed:
Task 2. WorkWithFile.
Edited code: method readFile rewrote.
Commit 26.12.2022 15:50
